managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java

```diff
@@ -415,6 +425,115 @@ private void createKubernetesInstanceTypes(Provider provider, UUID customerUUID)
     }
   }
 
+  // Performs discovery of region, zones, pull secret, storageClass
+  // when running inside a Kubernetes cluster. Returns the discovered
+  // information as a JSON, which is similar to the one which is
+  // passed to the createKubernetes method.
+  public Result getKubernetesConfigs(UUID customerUUID) {
+    try {
+      Map<String, Set<String>> regionToAZ = getKubernetesRegionToZoneInfo();
+      String storageClass = appConfig.getString("yb.kubernetes.storageClass");
+      String pullSecretName = appConfig.getString("yb.kubernetes.pullSecretName");
+      if (storageClass == null || pullSecretName == null) {
+        LOG.error("Required configuration keys from yb.kubernetes.* are missing.");
+        return ApiResponse.error(INTERNAL_SERVER_ERROR, "Required configuration is missing.");
```

Comment:
minor: throw YWServiceException instead. This is automatically handled by YWErrorHandler for all API requests.